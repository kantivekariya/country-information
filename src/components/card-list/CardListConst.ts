export const CARD_LIST_TEST_CONSTANT = {
  cardListTestId: "cardListTestId",
};
