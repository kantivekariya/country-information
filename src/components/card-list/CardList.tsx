import Card from "../card/Card";
import { CARD_LIST_TEST_CONSTANT } from "./CardListConst";

import "./_card-list.scss";

interface ICardListProps {
  countries: Array<{
    cca2: string;
    flags: { svg: string };
    name: {
      official: string;
    };
    population: number;
    region: string;
    capital: string;
  }>;
  theme: string;
}

const CardList = ({ countries, theme }: ICardListProps) => (
  <div
    className="cardList-container"
    data-testid={CARD_LIST_TEST_CONSTANT.cardListTestId}
  >
    {countries.map((country: any) => (
      <Card key={country.cca2} country={country} theme={theme} />
    ))}
  </div>
);

export default CardList;
