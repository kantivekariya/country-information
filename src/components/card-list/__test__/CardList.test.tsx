import { cleanup, render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import "@testing-library/jest-dom";

import CardList from "../CardList";
import { CARD_LIST_TEST_CONSTANT } from "../CardListConst";

afterEach(cleanup);

const cardListProps = {
  countries: [
    {
      cca2: "FI",
      flags: { svg: "https://flagcdn.com/fi.svg" },
      name: {
        official: "Republic of Finland",
      },
      population: 5530719,
      region: "Europe",
      capital: "Helsinki",
    },
    {
      cca2: "GT",
      flags: { svg: "https://flagcdn.com/gt.svg" },
      name: {
        official: "Republic of Guatemala",
      },
      population: 16858333,
      region: "Americas",
      capital: "Guatemala City",
    },
  ],
  theme: "light",
};

describe("Card List", () => {
  it("renders card list", () => {
    render(<CardList {...cardListProps} />, {
      wrapper: MemoryRouter,
    });
    const cardListElement = screen.getByTestId(
      CARD_LIST_TEST_CONSTANT.cardListTestId
    );
    expect(cardListElement).toBeInTheDocument();
    expect(cardListElement).not.toBeChecked();
  });

  it("should be matches snapshot", () => {
    const { baseElement } = render(<CardList {...cardListProps} />, {
      wrapper: MemoryRouter,
    });
    expect(baseElement).toMatchSnapshot();
  });
});
