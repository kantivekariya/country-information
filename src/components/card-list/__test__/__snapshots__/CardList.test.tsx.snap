// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card List should be matches snapshot 1`] = `
<body>
  <div>
    <div
      class="cardList-container"
      data-testid="cardListTestId"
    >
      <a
        class="link"
        data-testid="cardTestId"
        href="/FI"
      >
        <div
          class="card-container light"
        >
          <img
            alt="flag"
            class="card-flag"
            src="https://flagcdn.com/fi.svg"
          />
          <div
            class="card-description"
          >
            <h3
              class="card-title"
            >
              Republic of Finland
            </h3>
            <div
              class="card-text"
            >
              <span>
                Population: 
              </span>
              5,530,719
            </div>
            <div
              class="card-text"
            >
              <span>
                Region: 
              </span>
              Europe
            </div>
            <div
              class="card-text"
            >
              <span>
                Capital: 
              </span>
              Helsinki
            </div>
          </div>
        </div>
      </a>
      <a
        class="link"
        data-testid="cardTestId"
        href="/GT"
      >
        <div
          class="card-container light"
        >
          <img
            alt="flag"
            class="card-flag"
            src="https://flagcdn.com/gt.svg"
          />
          <div
            class="card-description"
          >
            <h3
              class="card-title"
            >
              Republic of Guatemala
            </h3>
            <div
              class="card-text"
            >
              <span>
                Population: 
              </span>
              16,858,333
            </div>
            <div
              class="card-text"
            >
              <span>
                Region: 
              </span>
              Americas
            </div>
            <div
              class="card-text"
            >
              <span>
                Capital: 
              </span>
              Guatemala City
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</body>
`;
